### PR TITLE
Rename jax.nn.normalize to standardize. Add normalize alias with DeprecationWarning

### LINKED ISSUE #9899: Rename jax.nn.normalize to jax.nn.standardize
As discussed: the jax.nn.normalize function actually standardizes by default, so we should name it accordingly.

Nobody reading `jax.nn.normalize(x)` expects the result not to have norm-1, and [PyTorch agrees with this too](https://pytorch.org/docs/stable/generated/torch.nn.functional.normalize.html).

The proposal here is just to rename to avoid user bugs. This would also open up the name `jax.nn.normalize` for a PyTorch-like implementation, but that should be as a follow-up, if desired.

### PR DESCRIPTION
**Pull request by dogeplusplus**
Fixes #9899 

Changes:
- `jax.nn.normalize` is renamed to `jax.nn.standardize`
- Add `normalize` as an alias with a `DeprecationWarning`
- Rename unit test to the new name along with references

Feel free to comment, change, or let me know if the deprecation should be in a separate PR.

### CODE REVIEW

File: jax/_src/nn/functions.py [hunk 1]
```diff
@@ -351,6 +352,16 @@ def normalize(x: Array,
         jnp.square(x), axis, keepdims=True, where=where) - jnp.square(mean)
   return (x - mean) * lax.rsqrt(variance + epsilon)
 
+@partial(jax.jit, static_argnames=("axis",))
```
Comment (froystig) [inline jax/_src/nn/functions.py:355]: We'll want to remove this `jit` decoration. It isn't needed since we call right into the jitted `standardize`, and it could interfere with issuing a warning (a side effect).

**Review by froystig — CHANGES_REQUESTED** [review]
Thanks for the PR!

Following the [API compatibility](https://jax.readthedocs.io/en/latest/api_compatibility.html) policy, could you announce the deprecation in [the changelog](https://github.com/google/jax/blob/main/CHANGELOG.md)? The jaxlib 0.3.2 changelog entry offers an example. Including a link to #9899 would be helpful.

**Review by froystig — APPROVED** [review]
Looks good. Thanks again!
